repos/spack_repo/builtin/packages/python/package.py

```diff
@@ -315,7 +315,8 @@ class Python(Package):
         depends_on("gmake", type="build")
         depends_on("pkgconfig@0.9.0:", type="build")
         depends_on("gettext +libxml2", when="+libxml2")
-        depends_on("gettext ~libxml2", when="~libxml2")
+        depends_on("iconv", when="~libxml2")
+        depends_on("gettext ~libxml2", when="~libxml2 ^[virtuals=iconv]gettext")
```

Comment:
```suggestion
        depends_on("gettext ~libxml2", when="~libxml2 %iconv=gettext")
```
some syntactic sugar